### PR TITLE
Update/Rename AttributePath/AttributePaths/AttributeStatus/AttributeStatuses/WriteResponseMessage, new tag(TagCompression)

### PR DESCRIPTION
**Pull request by yunhanw-google**
#### Problem
Update/Rename AttributePath/AttributePaths/AttributeStatus/AttributeStatuses/WriteResponseMessage with latest IM encoding spec
Minor new tag added: TagCompression in Path tag, currently it is disabled at default
#### Change overview
See above

#### Testing
The existing test covers, add unit test for TagCompression tag

### CODE REVIEW

**Comment by woody-apple** [comment]
@yunhanw-google does this also add a feature? (TagCompression) Hard to tell if it's new, or just a diff...

File: src/app/MessageDef/AttributePathIB.cpp [hunk 1]
```diff
@@ -0,0 +1,254 @@
+/**
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
```
Comment (yufengwangca) [inline src/app/MessageDef/AttributePathIB.cpp:3]: 2021

File: src/app/MessageDef/AttributePaths.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
+/**
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
```
Comment (yufengwangca) [inline src/app/MessageDef/AttributePaths.cpp:3]: 2021

File: src/app/MessageDef/AttributePathIB.cpp [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+ *    Copyright (c) 2018 Google LLC.
```
Comment (yufengwangca) [inline src/app/MessageDef/AttributePathIB.cpp:4]: Add  "Copyright (c) 2021 Project CHIP Authors"

**Review by yufengwangca — APPROVED** [review]

File: src/app/MessageDef/AttributePathIB.cpp (hunk 3 above)
Comment (yunhanw-google) [reply src/app/MessageDef/AttributePathIB.cpp:4]: Done, thanks

File: src/app/MessageDef/AttributePaths.cpp (hunk 2 above)
Comment (yunhanw-google) [reply src/app/MessageDef/AttributePaths.cpp]: Done, thanks

File: src/app/MessageDef/AttributePathIB.cpp (hunk 1 above)
Comment (yunhanw-google) [reply src/app/MessageDef/AttributePathIB.cpp]: Done, thanks

**Review by LuDuda — APPROVED** [review]
LGTM :+1:

**Comment by woody-apple** [comment]
Looking through this diff, it does indeed add a feature. @yunhanw-google can you update the description and title please? This isn't a simple rename.

@bzbarsky-apple can you take a peek?

**Comment by yunhanw-google** [comment]
> Looking through this diff, it does indeed add a feature. @yunhanw-google can you update the description and title please? This isn't a simple rename.
> 
> @bzbarsky-apple can you take a peek?

updated

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**